# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/script/ModuleLoader.cpp
# Commit: 88588284e643
# Full Hash: 88588284e64389c983816d5d61c9834fa7a14e40
# Author: Yoshi Cheng-Hao Huang <allstars.chh@gmail.com>
# Date: 2025-09-26 06:07 +0000
# Regressor Bug: 1968890
# File Overlap Count: 8
# Overlapping Files: dom/script/ScriptLoadContext.cpp, dom/script/ModuleLoader.cpp, js/loader/ModuleLoadRequest.cpp, js/loader/ModuleLoaderBase.cpp, dom/workers/loader/WorkerModuleLoader.cpp
# Description:
#   Bug 1968890 - Part 1: Dynamic import is not a top-level request. r=jonco,dom-worker-reviewers,edenchuang
# ==============================================================================

diff -r b5be4eed1485 -r 88588284e643 dom/script/ModuleLoader.cpp
--- a/dom/script/ModuleLoader.cpp	Fri Sep 26 09:40:21 2025 +0300
+++ b/dom/script/ModuleLoader.cpp	Fri Sep 26 06:07:43 2025 +0000
@@ -164,7 +164,7 @@
 void ModuleLoader::OnModuleLoadComplete(ModuleLoadRequest* aRequest) {
   MOZ_ASSERT(aRequest->IsFinished());
 
-  if (aRequest->IsTopLevel()) {
+  if (aRequest->IsTopLevel() || aRequest->IsDynamicImport()) {
     if (aRequest->GetScriptLoadContext()->mIsInline &&
         aRequest->GetScriptLoadContext()->GetParserCreated() ==
             NOT_FROM_PARSER) {